LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/BasicsStationNetworkServerStartup.cs

```diff
@@ -66,6 +69,20 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             // TO DO: When certificate generation is properly handled, enable https redirection
             // app.UseHttpsRedirection();
 
+            // We want to make sure the module connection is started at the start of the Network server.
+            // This is needed when we run as module, therefore we are blocking.
+            if (networkServerConfiguration.RunningAsIoTEdgeModule)
```

Comment:
What about using real async for this?

I would remove the CreateAsync method invocation from here and place it in the BasicsStationNetworkServer.RunServerAsync

We could, after WebHost build, do the following:

    using (var scope = webHost.Services.CreateScope())
    {
        // We want to make sure the module connection is started at the start of the Network server.
        // This is needed when we run as module, therefore we are blocking.
        if (scope.ServiceProvider.GetRequiredService<NetworkServerConfiguration>().RunningAsIoTEdgeModule)
        {
            var moduleConnection = scope.ServiceProvider.GetRequiredService<ModuleConnectionHost>();
            await moduleConnection.CreateAsync();
        }
    }

At this stage, we could also pass the CancellationToken provided to RunServerAsync method

Reply:
@atifaziz What do you think about Daniele suggestion? ^^

Reply:
It's a good suggestion. Do we need the scope? If so, I would create it if needed:

```c#
if (webHost.Services.GetRequiredService<NetworkServerConfiguration>().RunningAsIoTEdgeModule)
{
    using var scope = webHost.Services.CreateScope();
    await scope.ServiceProvider.GetRequiredService<ModuleConnectionHost>().CreateAsync();
}
```